scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -468,58 +515,68 @@ We can correct this by applying a "shrinkage" procedure, which will adjust large
+Although some of us may have all of the identifiers memorized by heart, it can be useful to have a human readable symbol in our results.
+Before we save the results as a file, we will grab the gene symbols from the `rowData` of our original SCE object and add them as a new column.
 
-```{r}
-# create a data frame with the results 
-deseq_results <- shrink_results %>%
+```{r add gene symbol}
+# convert shrunken results to a data.frame and add ensembl id as a column
+deseq_results <- shrink_results |>
   tibble::as_tibble(rownames = "ensembl_id")
 
-# first look at the significant results 
-deseq_results_sig <- deseq_results %>%
-  dplyr::filter(padj <= 0.05)
-
-deseq_results_sig
-```
-
-
-```{r}
-# convert rowdata to dataframe 
-sce_rowdata_df <- rowData(tumor_myoblast_sce) %>%
+# convert rowdata to data.frame 
+sce_rowdata_df <- rowData(tumor_myoblast_sce) |>
   # create a column with rownames stored as ensembl id to use for joining with deseq results
   tibble::as_tibble(rownames = "ensembl_id")
 
 # combine deseq results with rowdata by ensembl id 
-deseq_results <- deseq_results %>%
+deseq_results <- deseq_results |>
   dplyr::left_join(sce_rowdata_df, by = "ensembl_id")
 
 head(deseq_results)
 ```
 
-```{r}
-# save our results 
+We can save the new `data.frame` that we have created with the ensembl identifiers, gene symbols, and the `DESeq2` results as a tab separated (`tsv`) file.
+
+```{r save deseq}
+# save our results as tsv
 readr::write_tsv(deseq_results, deseq_output_file)
 ```
 
+The last thing that we will do is take a look at how many genes are significant.
+Here we will want to use the adjusted p-value, found in the `padj` column of the results, as this p-value accounts for multiple test correction.
 
-### Exploring the identified differentially expressed genes 
+```{r}
+# first look at the significant results 
+deseq_results_sig <- deseq_results |>
+  # filter based on adjusted pvalue
+  dplyr::filter(padj <= 0.05)
 
-Let's also do some initial analysis to look at what types of genes are being identified and see if what is being identified as actually up/down regulated appear to correlate back to the single cell data. 
+head(deseq_results_sig)
+```
 
-The first plot we'll make is a volcano plot and we can label the significant genes with their gene symbols. 
 
-```{r}
+### Exploring the identified differentially expressed genes 
+
+Now that we have identified a set of genes that are DE in the tumor myoblasts between ARMS and ERMS subtypes, lets actually take a look at them and see if we can make some informative plots.
+The first plot we'll make is a volcano plot using the [`EnhancedVolcano` package](https://github.com/kevinblighe/EnhancedVolcano).
+This package automatically color codes the points by cutoffs for both significance and fold change and labels many of the significant genes (subject to spacing).
+`EnhancedVolcano` has many, many options, which is a good thing if you don't like all of it's default settings.
```

Comment:
Putting this comment here, because I can't figure out where else, but I feel like the "fun arrows" point the wrong way. Nothing to be done about it.